feature/#619 more flexible CacheManager

monsieurtanuki opened the pull request:
* More flexible CacheManager
* Slight impact of CacheManager refactoring on the other 2 classes
* Slight correction on the manifest not related to #619 but needed when testing

secaw commented:
This is such a great improvement - now we can also download tile overlays like OpenSeaMap together with the base map.
Tested this & works.
Thank You!

spyhunter99 commented on osmdroid-android/src/main/java/org/osmdroid/tileprovider/cachemanager/CacheManager.java:
```diff
@@ -61,24 +63,31 @@
+        this(mapView.getTileProvider(), writer, mapView.getMinZoomLevel(), mapView.getMaxZoomLevel());
+    }
+
+    public CacheManager(final MapTileProviderBase pTileProvider,
```
since this is a new constructor, please add a javadoc comment that references the issue for the PR and an @ since 5.6.5

spyhunter99 commented:
@monsieurtanuki mentioned in this issue, but not in the code (?) was the addition of a new constructor for Cache Manager. Was that added or am i just missing it, or is this a 'push to another issue' task?

other than this and the minor javadoc issue, i don't see any obvious problems

monsieurtanuki commented:
@spyhunter99 I just added a slight Javadoc to the new `CacheManager` constructor. In order to clear the confusion, I'm talking about the constructor referred to in the issue description and added in the previous code push.
